pkg/apis/eventing/v1beta3/eventtype_validation.go

```diff
@@ -60,3 +56,30 @@ func (et *EventType) CheckImmutableFields(ctx context.Context, original *EventTy
 	}
 	return nil
 }
+
+func (ets *EventTypeSpec) ValidateAttributes() *apis.FieldError {
+	attributes := make(map[string]EventAttributeDefinition, len(ets.Attributes))
+	for _, attr := range ets.Attributes {
+		attributes[attr.Name] = attr
+	}
+
+	missingFields := []string{}
+	if _, ok := attributes["type"]; !ok {
+		missingFields = append(missingFields, "type")
+	}
+	if _, ok := attributes["source"]; !ok {
+		missingFields = append(missingFields, "source")
+	}
+	if _, ok := attributes["specversion"]; !ok {
+		missingFields = append(missingFields, "specversion")
+	}
+	if _, ok := attributes["id"]; !ok {
+		missingFields = append(missingFields, "id")
+	}
```

Comment:
I'm wondering that to make the usage easier we could have id and `specversion` be optional even if it's technically required

Reply:
but I think starting with required is fine as making it optional would be a non-breaking change while the opposite is not true

Reply:
That makes sense to me, especially if `id` is going to just be a UUID, and specversion will _probably_ just be 1.0